specs/sds_detection_modules.adoc

```diff
@@ -0,0 +1,260 @@
+// vim: tw=79
+
+= SDS detection modules in node_agent
+
+Node agent serves as the component that is responsible for invoking and/or
+executing operations that are part of global Tendrl scope.
+
+Tendrl officially supports specific storage systems (ceph and gluster). However
+Tendrl also aims to be a framework that provides some essential capabilities
+and interfaces so that it is easy to add support for storage systems that are
+not officially supported.
+
+For this Tendrl should be able to auto-detect existing deployments and out of
+the box sane defaults for laying out the storage system and managing them.
+
+== Problem description
+
+Tendrl need to provide a pluggable model using which auto-detection modules
+for different storage systems can be deployed in pre-defined locations and
+Tendrl can load and execute certain generic APIs on these modules to figure the
+underlying pre-existing storage system. This can become the pre-requisite which
+gets executed as part of starting node agent on the storage nodes and helps
+import cluster flow in Tendrl to decide what kind of cluster is getting imported
+. This also would help deciding on deployment of corresponding Tendrl
+integration module for managing the cluster seamlessly.
+
+== Use Cases
+
+This specification covers the import cluster flow in Tendrl and auto detection
+of underlying storage systems on storage nodes.
+
+== Proposed change
+
+We try to introduce a pluggable set of modules (for officially supported storage
+systems from Tendrl) which could deployed in a standard Tendrl defined location
+and would be executed as part of node agent start on nodes. These modules would
+auto detect the underlying storage systems and its related details. These
+details would be tagged with storage nodes so that while import cluster Tendrl
+would be able to decide on deployment of corresponding integration modules on
+the storage systems.
+
+A summary of proposed changes are
+
+* Define a standard path from where all the available storage detection modules
+would be loaded
+
+* While starting the node agent service on the storage nodes, execute these
+plugins and auto detect the storage systems (if any) on underlying storage nodes
+
+* Tag the dectected details (if any) with the storage nodes details
+
+* Enhance the import cluster flows in node agent component to decide on what
+Tendrl integration module gets deployed on the underlying storage nodes
+
+=== Alternatives
+
+None
+
+=== Data model impact:
+
+None
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+* API layer should attach the storage system specific auto-detected details of
```

Comment:
are you planning to do this as part of this spec. i feel limit the scope this spec only to detection of cluster details and updating the etcd